pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/json/MutableJsonIndexImpl.java

```diff
@@ -367,10 +379,18 @@ public void convertFlattenedDocIdsToDocIds(Map<String, RoaringBitmap> valueToFla
   }
 
   @Override
-  public Map<String, RoaringBitmap> getMatchingFlattenedDocsMap(String jsonPathKey) {
+  public Map<String, RoaringBitmap> getMatchingFlattenedDocsMap(String jsonPathKey, @Nullable String filterString) {
```

Comment:
Not introduced in this PR, but I think we should support json path key with and without `$` prefix. See `getMatchingFlattenedDocIds()` for reference

Reply:
Added the support